ksql-streams/src/main/java/io/confluent/ksql/execution/streams/ExecutionStepFactory.java

```diff
@@ -215,14 +215,15 @@ private ExecutionStepFactory() {
     );
   }
 
-  public static <K> StreamSelectKey<KStream<K, GenericRow>> streamSelectKey(
+  @SuppressWarnings("unchecked")
+  public static <K> StreamSelectKey<KStream<Struct, GenericRow>> streamSelectKey(
       final QueryContext.Stacker stacker,
       final ExecutionStep<KStream<K, GenericRow>> source,
       final String fieldName,
       final boolean updateRowKey
   ) {
     final QueryContext queryContext = stacker.getQueryContext();
-    return new StreamSelectKey<>(
+    return (StreamSelectKey<KStream<Struct, GenericRow>>) new StreamSelectKey(
```

Comment:
use of raw type.

We need to get the generics right here.

Will `StreamSelectKey` always build a `KStream<Struct, GenericRow>` regardless of whether it's source is `KStream<Windowed<Struct>, GenericRow>` or `KStream<Struct, GenericRow>`??? I think it does, right?

That being the case, `StreamSelectKey` should be defined as:

```java
public class StreamSelectKey implements ExecutionStep<KStream<Struct, GenericRow>> {

  private final ExecutionStep<KStream<?, GenericRow>> source;
  ...

  public StreamSelectKey(
      ...
      final ExecutionStep<KStream<?, GenericRow>> source,
     ...
  )

  ...
}
```

And this method should be:

```java
public static StreamSelectKey streamSelectKey(
      final QueryContext.Stacker stacker,
      final ExecutionStep<KStream<?, GenericRow>> source,
      final String fieldName,
      final boolean updateRowKey
  ) {
    final QueryContext queryContext = stacker.getQueryContext();
    return new StreamSelectKey(
        new DefaultExecutionStepProperties(
            source.getProperties().getSchema(),
            queryContext
        ),
        source,
        fieldName,
        updateRowKey
    );
  }
```